src/server/transaction/cmds/cmds.go

```diff
@@ -92,18 +92,28 @@ transaction' or cancelled with 'delete transaction'.`,
 				return err
 			}
 			defer c.Close()
-			transaction, err := c.StartTransaction()
-			if err != nil {
-				return grpcutil.ScrubGRPC(err)
-			}
-			// TODO: use advisory locks on config so we don't have a race condition if
-			// two commands are run simultaneously
-			err = setActiveTransaction(transaction)
+			txn, err := getActiveTransaction()
 			if err != nil {
 				return err
 			}
```

Comment:
nit: it's a bit more conventional in go to minimize nesting and avoid else blocks, i.e. do this instead:

```go
if txn != nil {
    return errors.Errorf("cannot start a new transaction, since transaction with ID %q already exists", txn.ID)
}
```

Also note a couple of other changes in that code:

1) Errors messages are lowercased, I'm not sure if this is a go convention or just a convention we follow though
2) Use `%q` to format `txn.ID`, so it'll be surrounded by quotes

Reply:
hey thanks, I didn't know about this convention.

Reply:
hey thanks, I didn't know about this convention. Will keep in mind :smile: